packages/common/core-utils/src/events/index.ts

```diff
@@ -17,3 +17,12 @@ export {
 	type Off,
 	type IsListener,
 } from "./listeners.js";
+
+export {
+	type UnionToIntersection,
+	type MapGetSet,
+	type NestedMap,
+	getOrAddInMap,
+	setInNestedMap,
+	getOrCreate,
```

Comment:
I think if you change emitter.ts to eliminate getOrCreate usage and get rid of interop.ts, then all of utils can remain in the tree package as-is.

Reply:
`getOrCreate` is used by `EventEmitter` class  and experimental/tree: https://github.com/microsoft/FluidFramework/blob/main/experimental/dds/tree/src/Common.ts#L308. 

probably best to move `getOrCreate` to core-utils and rest of the functions (`getOtAddInMap, setInNestedMap`) can remain in tree?

Reply:
I'd suggest either:
1. Remove getOrCreate usage from emitter.ts, and instead just implement the logic locally
2. Duplicate getOrCreate in emitter.ts and do not export it from core-utils

getOrCreate may be end up being generally useful and we may eventually want to make it a global utility, but let's defer that unless/until we make an explicit decision to do so.  That choice should probably be paired with a survey of the code base to uniformly use it (both for bundle size and code consistency).  Or as it's more commonly stated, even though this is a tiny utility it should still follow the rule of three :)